javaslang/src/main/java/javaslang/collection/Queue.java

```diff
@@ -40,7 +40,7 @@
  * @author Daniel Dietrich
  * @since 2.0.0
  */
-public class Queue<T> implements Kind1<Queue<?>, T>, LinearSeq<T>, Serializable {
+public class Queue<T> extends AbstractsQueue<T, Queue<T>> implements Serializable {
```

Comment:
Will add them back and add an `@Override` :) (in `3.0` we can simply delete the overridden impls)